### PR TITLE
Docs/openshift provider

### PR DESCRIPTION
**Pull request by goern**
added a human readable description of provider's features and configuration options.

### CODE REVIEW

File: Providers.asciidoc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+Not all Atomic Apps must support each provider: one Atomic App may
+only include deployment information for OpenShift3 while another does
+support OpenShift3 and Kubernets.
+
+=== List of providers
+
+Atomic App 0.0.2 includes three providers: 
+
+* Docker, which is the most simple one
+* Kubernets
```
Comment (aweiteka) [inline Providers.asciidoc]: typo: Kubernetes

File: Providers.asciidoc [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+While deploying and Atomic App you can choose one of the providers by
+setting it in `answers.conf`:
+
+```
+{
+    "general": {
+            "provider": "openshift"
+    }
+}
+```
```
Comment (aweiteka) [inline Providers.asciidoc]: I think an ini-style conf example is preferred:

```
[general]
provider: openshift
```

File: Providers.asciidoc [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+Providers may need additional configuration.
+
```
Comment (aweiteka) [inline Providers.asciidoc]: Could you add a reference to Docker provider? Even as a placeholder would be better than leaving it out altogether.

File: Providers.asciidoc [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+with Atomic App. It gives you a short overview of all available
+providers and how to use them.
+
+Not all Atomic Apps must support each provider: one Atomic App may
+only include deployment information for OpenShift3 while another does
+support OpenShift3 and Kubernets.
+
+=== List of providers
+
+Atomic App 0.0.2 includes three providers:
```
Comment (vpavlin) [inline Providers.asciidoc]: Should be 0.1.0 I guess

File: Providers.asciidoc [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+* Kubernets
+* OpenShift 3
+
+All providers assume that you install and run the Atomic App on a host
+that is part of the backend cluster or runs docker directly. By now we
+do not support remote deployments. (see also )
+
+=== Choosing a provider
+
+While deploying and Atomic App you can choose one of the providers by
```
Comment (vpavlin) [inline Providers.asciidoc]: type: deploying an Atomic App

File: Providers.asciidoc [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+application into a given namespace. All components of the applications
+will be created in that namespace.
+
+==== configuration values
+
+.Kubernetes Provider configuration values
+[frame="topbot",grid="none",options="header",cols="3,1,10,5"]
+|====================================================
+|keyword |required |description |default value
+|namespace|no|a namespace to use with each kubectl call|none
```
Comment (vpavlin) [inline Providers.asciidoc]: actually the default value is `default`

File: Providers.asciidoc [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+=== OpenShift3 Provider
+
+==== features
+
+The OpenShift3 Provider will deploy and run an Atomic App on a
+OpenShift instance provided via a OpenShift configuration file.
+And OpenShift configuration file is written to disk after you have
+logged in, see
+http://lists.openshift.redhat.com/openshift-archives/users/2015-March/msg00014.html[osc
+login annoncement]
```
Comment (vpavlin) [inline Providers.asciidoc]: typo: announcement

File: Providers.asciidoc [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+You need to provide a copy of `.config/openshift/.config` so that the
+provider may use this configuration to deploy and run the Atomic App.
+
+As of 0.0.2 of Atomic App, only
```
Comment (vpavlin) [inline Providers.asciidoc]: version check, please..

File: Providers.asciidoc [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+template file will be replaced by the
+https://github.com/projectatomic/nulecule/blob/master/spec/0.0.2/README.md#paramsObject[params]
+supplied to Atomic App, names need to match.
+
+==== configuration values
+
+.OpenShift3 Provider configuration values
+[frame="topbot",grid="none",options="header",cols="3,1,10,5"]
+|====================================================
+|keyword |required |description |default value
```
Comment (vpavlin) [inline Providers.asciidoc]: What is this?

Comment (vpavlin) [inline Providers.asciidoc]: Eh..sorry, never mind...I got a bit confused..as always..

**Pull request merged**